SCClassLibrary/Common/Math/Date.sc

```diff
@@ -5,12 +5,24 @@ Date {
 	*getDate { ^this.localtime }
 	*localtime { ^this.new.localtime }
 	*gmtime { ^this.new.gmtime }
-	*new { arg year, month, day, hour=0, minute=0, second=0, dayOfWeek, rawSeconds;
+	*new { arg year, month, day, hour, minute, second, dayOfWeek, rawSeconds;
 		var instance = super.newCopyArgs(year, month, day, hour, minute, second, dayOfWeek,
 			rawSeconds);
-		if (dayOfWeek.isNil or: {rawSeconds.isNil}) {
-			instance.prResolve;
+		case
```

Comment:
somehow, I'd prefer if `resolve` would take care of all those case distinctions. Then they wouln't have to clutter the `*new` implementation.

Reply:
Do you mean just moving this logic into the Date class `resolve` instance method, or into the C++ helper function?

Reply:
Actually, I wanted `resolve` to always resolve from YMD:HMS -> rawSeconds (there is currently a private helper `prResolveFromRawSeconds` to go the other way, rawSeconds -> YMD:HMS).  In discussion with @brianlheim we agreed that when one needed to convert *from* rawSeconds, it was acceptable to create a new Date instance, passing only the rawSeconds argument.

So if I move the current `*new` logic into `resolve`, then it won't work properly when you call `resolve` on an existing Date object (which will have a filled-in rawSeconds variable).  In this case, you'd want it to recalculate rawSeconds from YMD:HMS, but it would do the opposite.  So I want to be able to distinguish between creating a new Date vs requesting `resolve` on an existing instance.

However, I guess we **could** require the user to manually "nil" out the rawSeconds field (on an existing Date instance) if they want this behaviour, then we could have it all be done by `resolve`.  If you prefer that, I would change it so that:

1. When creating a new instance (and calling `resolve` on an existing instance), it would use the rawSeconds field if provided, overwriting everything else.
2. Calling `resolve` on an existing instance would "resolve from rawSeconds" (rawSeconds -> YMD:HMS).  e.g. If you did: `d.year = d.year+1; d.resolve;` you would lose your modified year and go back to the original date, because everything would be recomputed from rawSeconds.
3. When you want to do the reverse (YMD:HMS -> rawSeconds), you would have to do: `d.year = d.year+1; d.rawSeconds = nil; d.resolve;`

I'm not sure this is the best idea, though, because my intuition tells me one will more often want to manipulate Dates in the YMD:HMS format, and the rawSeconds would be mainly used internally (for comparing Dates, etc).

Hmm... An alternative approach would be to have `*new` call `prResolveFromRawSeconds` instead, which would be modified to resolve from rawSeconds if it exists, but otherwise perform regular `resolve` (and return `localtime` if everything is unspecified).  Then the current logic of the `resolve` method could be left alone, and `*new` would also get rid of this cluttery logic you don't like.  How's that sound?

Reply:
In the end it makes more sense to just add an initialization helper method.  I did that, and updated my terminology to use the correct term ("instance variables").  See [here](https://github.com/supercollider/supercollider/pull/3259/commits/30644b0).